### PR TITLE
feat: faster "Add New Person" user experience via a couple of tweaks

### LINKED ISSUE #3425: Faster "Add New Person" user experience via a couple of tweaks
This is a pair of possibly small feature requests. I noticed that when we search for a contact in the main contact bar, Monica will only display the "Add New Person" option if there were no matches at all against the current database of contacts. If the person I'm searching for doesn't exist yet but shares the same name/substring with others that already exist, the "Add New Person" link doesn't appear. This is frustrating as I'll have to add the person by moving over to "Add Someone" and then fill all the information again.

Here's a snippet of typing Pedr on my current database:
image

Now if I type Pedri, the Add New Person option appears:
image

Also, when I do click on "Add New Person" or "Add Someone", it brings me to a blank form.

**Describe the solution you'd like**
I'd like you to consider two changes:
- First, if I choose "add new person" from the drop down, consider copying the current text from the search bar over - if a single string is there, consider that the first name. If two strings separated by a space, prefill as first, last name.

Something like:
image
would become
image

- The other suggestion is to leave the "Add New Person" dialog option as an option on the drop down of names, even if substrings have been found.

**Describe alternatives you've considered (optional)**
No alternatives at this point.

**Additional context**
First feature request, apologies if this is not the right way to engage with the project.

Merci!!!

### PR DESCRIPTION
**Pull request by jeremies**
Close #3425

### CODE REVIEW

File: resources/js/components/people/partials/ContactAutosuggest.vue [hunk 1]
```diff
@@ -164,13 +164,6 @@ export default {
       var datas = this.cache[text];
 
       datas = datas.filter(this.filter);
-      if (datas.length === 0) {
```
Comment (jeremies) [inline resources/js/components/people/partials/ContactAutosuggest.vue:167]: dead code

File: resources/js/components/people/ContactSearch.vue [hunk 2]
```diff
@@ -49,7 +49,19 @@ export default {
...
-        window.location = 'people/add';
+        const names = contact.item.keyword.split(' ').map(name =>
+          name.charAt(0).toUpperCase() + name.substr(1).toLowerCase()
+        );
+        const first_name = names[0];
+        const last_name = names.slice(1).join(' ');
+        
+        let params = new URLSearchParams();
+        if (first_name)
+          params.set('first_name', first_name);
```
Comment (asbiin) [inline resources/js/components/people/ContactSearch.vue:60]:
```suggestion
        if (first_name) {
          params.set('first_name', first_name);
        }
```

File: resources/js/components/people/ContactSearch.vue [hunk 3]
```diff
@@ -49,7 +49,19 @@ export default {
...
+        if (last_name)
+          params.set('last_name', last_name);
```
Comment (asbiin) [inline resources/js/components/people/ContactSearch.vue:62]:
```suggestion
        if (last_name) {
          params.set('last_name', last_name);
        }
```

File: resources/js/components/people/ContactSearch.vue [hunk 4]
```diff
@@ -49,7 +49,19 @@ export default {
       if (contact.item.id > 0) {
         window.location = contact.item.route;
       } else {
-        window.location = 'people/add';
+        const names = contact.item.keyword.split(' ').map(name =>
+          name.charAt(0).toUpperCase() + name.substr(1).toLowerCase()
```
Comment (asbiin) [inline resources/js/components/people/ContactSearch.vue:53]: You might use `_.capitalize`. See https://lodash.com/docs/4.17.15#capitalize

File: resources/js/components/people/ContactSearch.vue [hunk 5]
```diff
@@ -49,7 +49,19 @@ export default {
...
+        );
+        const first_name = names[0];
```
Comment (asbiin) [inline resources/js/components/people/ContactSearch.vue:55]: Everybody don't use "first_name last_name", actually there is a config for that in the settings page.

File: resources/js/components/people/ContactSearch.vue (hunk 2 above)
Comment (jeremies) [reply resources/js/components/people/ContactSearch.vue]: In the last commit this is resolved

File: resources/js/components/people/ContactSearch.vue (hunk 3 above)
Comment (jeremies) [reply resources/js/components/people/ContactSearch.vue]: In the last commit this is resolved

File: resources/js/components/people/ContactSearch.vue (hunk 4 above)
Comment (jeremies) [reply resources/js/components/people/ContactSearch.vue]: In the last commit this is resolved

File: resources/js/components/people/ContactSearch.vue (hunk 5 above)
Comment (jeremies) [reply resources/js/components/people/ContactSearch.vue]: In the last commit this is resolved

**Comment by asbiin** [comment]
@jeremies Thank you for your contribution!

**Pull request merged**